apollo-router-core/src/query_planner/mod.rs

```diff
@@ -84,8 +79,8 @@ impl PlanNode {
     fn execute_recursively<'a>(
         &'a self,
         current_dir: &'a Path,
-        request: &'a Request,
-        service_registry: &'a dyn ServiceRegistry,
+        context: &'a Arc<RwLock<Context>>,
```

Comment:
Do we need this RwLock? I can't find anywhere we take a write() on it.

Reply:
we will need it when aggregating Cache-Control headers from subgraph responses